Make App parent command to exentd from BaseCommand to expose --help option

andsel opened the pull request:
<!--
Thanks for submitting your Pull Request!

Please delete this text, and add description of the topic solved by this PR.

To make releases as automated as possible we use conventional commits formats.
Thus commits and pull-requests titles should before merge follow the Conventional Commits spec.
See https://www.conventionalcommits.org/

Details in https://github.com/Ezard/semantic-prs

If in doubt then open the pull-request and we'll help you - Thank you!
-->

maxandersen commented:
you need to run formatting - `just format` or `./gradlew spotlessApply`

quintesse commented:
@andsel so actually to _real_ change that needs to be made is this:

```
@CommandLine.Command(name = "app", description = "Manage scripts installed on the user's PATH as commands.", subcommands = {
   AppInstall.class, AppList.class, AppUninstall.class, AppSetup.class },

--> mixinStandardHelpOptions = true)
```

If you add that attribute to the annotation it will just work.

Extra points if you fix it for the other commands as well :-)

andsel commented:
Hi @quintesse I've added a couple of commits to recreate what you described and added a couple of test just for the `app` top command.
Let me know what do you think, so that I can replicate this in all the other top commands that has subcommands.

andsel commented:
yes, I'll do in this PR if it's ok for you.

andsel commented:
I've found that also `cache` (clear) `jdk` (list, install, uninstall, home, java-env, default) `trust` (add, remove) `wrapper` (install) have sub-commands, so I have to validate also those.

andsel commented:
Thanks @quintesse , now it's ready to get reviewed :-)

quintesse reviewed (approved):
Nice job @andsel !
